sphinx/util/fileutil.py

```diff
@@ -4,17 +4,33 @@
 
 import os
 import posixpath
+import warnings
 from typing import TYPE_CHECKING, Callable
 
 from docutils.utils import relative_path
 
+from sphinx.deprecation import RemovedInSphinx80Warning
 from sphinx.util.osutil import copyfile, ensuredir
 from sphinx.util.typing import PathMatcher
 
 if TYPE_CHECKING:
     from sphinx.util.template import BaseRenderer
 
 
+def _template_basename(filename: str) -> str | None:
+    # TODO: remove this method
+    if filename.lower().endswith('_t'):
+        warnings.warn(
+            f"{filename!r}: filename suffix '_t' for templates is deprecated. "
+            "If the file is a Jinja2 template, use the suffix '.jinja' instead.",
+            RemovedInSphinx80Warning,
```

Comment:
Perhaps a generic ``PendingDeprecationWarning``, with a comment to re-visit not earlier than (e.g.) 1 Jan 2025?